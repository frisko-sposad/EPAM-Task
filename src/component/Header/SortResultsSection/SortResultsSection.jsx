import React from 'react';
import './SortResultsSection.css';

const SortResultsSection = ({ filmsBy, genre, moviesFound, sortBy, releaseDate, rating }) => (
  <section className="result-sort__container">
    {(filmsBy || moviesFound) && (
      <div>
        <span>
          <strong>
            {filmsBy}
            {moviesFound}
          </strong>
        </span>
        {genre && <span>{genre}</span>}
      </div>
    )}
    {sortBy && (
      <div>
        <span>
          <strong>{sortBy}</strong>
        </span>
        <button type="button" className="btn_sort">
          {releaseDate}
        </button>
        <button type="button" className="btn_sort">
          {rating}
        </button>
      </div>
    )}
  </section>
);

export default SortResultsSection;
