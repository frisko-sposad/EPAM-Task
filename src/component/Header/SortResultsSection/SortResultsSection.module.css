.header__container {
  color: white;
  padding: 1rem;
  background-image: linear-gradient(
      rgba(0, 0, 0, 0.8) 0%,
      rgba(0, 0, 0, 0.8) 100%
    ),
    url(https://assets.nflxext.com/ffe/siteui/vlv3/70731dc8-8977-4ed7-9a0a-421f510aea00/8cbb059e-e3a8-4e36-80fe-33b90a88b39f/RU-en-20200914-popsignuptwoweeks-perspective_alpha_website_large.jpg);
}

.resultSort__container {
  display: flex;
  justify-content: space-between;
  padding: 1rem 3rem;
  font-size: 1rem;
  font-weight: 400;
  text-decoration: none;
  background-color: #dddddd;
}

.sortBtn {
  text-decoration: none;
  color: #000000;
  background: transparent;
  padding: 0 1rem;
  border: none;
  outline: none;
  font-size: 1rem;
  cursor: pointer;
}

.sortBtn:hover {
  opacity: 0.5;
}

.sortBtn:active {
  opacity: 0.8;
}

.sortBtn:focus {
  color: tomato;
  font-weight: 600;
}
