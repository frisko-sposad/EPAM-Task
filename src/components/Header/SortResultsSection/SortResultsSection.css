.result-sort__container {
  display: flex;
  justify-content: space-between;
  padding: 1rem 3rem;
  font-size: 1rem;
  font-weight: 400;
  text-decoration: none;
  background-color: #dddddd;
}
