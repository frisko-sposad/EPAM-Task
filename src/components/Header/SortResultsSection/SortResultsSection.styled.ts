import styled from 'styled-components';

const SortResultContainer = styled.h1`
  height: 1.2rem;
  display: flex;
  justify-content: space-between;
  padding: 1rem 3rem;
  font-size: 1rem;
  font-weight: 400;
  text-decoration: none;
  background-color: #dddddd;
  color: #000000;
`;

export default SortResultContainer;
