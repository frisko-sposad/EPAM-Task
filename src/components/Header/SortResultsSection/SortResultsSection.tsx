import React, { FC, useCallback } from 'react';
import { useRouter } from 'next/router';
import Button from '../../Generic/Button/Button';
import SortResultContainer from './SortResultsSection.styled';
import VariantBtn from '../../Generic/Button/Button.types';

interface SortResultsSectionProps {
  isSearchPage?: boolean;
  genres: string[] | undefined;
  moviesFound?: number;
}

const SortResultsSection: FC<SortResultsSectionProps> = ({ isSearchPage, genres, moviesFound }) => {
  const router = useRouter();
  const sortByParam = router?.query.sortBy;
  const searchParam = router?.query.search;
  const searchByParam = router?.query.searchBy;

  const handleSortMoviesClick = useCallback(
    (option) => () => {
      router.push(`/search/?sortBy=${option ?? 'release_date'}&search=${searchParam}&searchBy=${searchByParam}`);
    },
    [router, searchParam, searchByParam],
  );

  return (
    <SortResultContainer>
      {isSearchPage ? (
        <>
          {moviesFound !== 0 && moviesFound && (
            <>
              <span>
                <strong>{`${moviesFound} movies found`}</strong>
              </span>
              <div>
                <span>
                  <strong>Sort by: </strong>
                </span>
                <Button
                  variant={VariantBtn.SortBy}
                  active={sortByParam === 'release_date'}
                  onClick={handleSortMoviesClick('release_date')}
                >
                  release date
                </Button>
                <Button
                  variant={VariantBtn.SortBy}
                  active={sortByParam === 'vote_average'}
                  onClick={handleSortMoviesClick('vote_average')}
                >
                  rating
                </Button>
              </div>
            </>
          )}
        </>
      ) : (
        <div>
          <span>
            <strong>Films by: </strong>
          </span>
          {genres && <span>{genres[0]} genre</span>}
        </div>
      )}
    </SortResultContainer>
  );
};

export default SortResultsSection;
