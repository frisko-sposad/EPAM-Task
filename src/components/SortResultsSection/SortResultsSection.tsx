import React, { FC, memo, useCallback } from 'react';
import { useRouter } from 'next/router';
import Button from '../Generic/Button/Button';
import SortResultContainer from './SortResultsSection.styled';
import VariantBtn from '../Generic/Button/Button.types';

interface SortResultsSectionProps {
  isSearchPage: boolean;
  genres?: string[];
  moviesCount: number;
}

const SortResultsSection: FC<SortResultsSectionProps> = ({ isSearchPage, genres, moviesCount }) => {
  const router = useRouter();
  const { sortBy, search, searchBy } = router?.query?.sortBy
    ? router?.query
    : {
        sortBy: 'release_date',
        search: '',
        searchBy: 'title',
      };

  const handleSortMoviesClickByReleaseDate = useCallback(() => {
    router.push(`/search/?sortBy=release_date&search=${search}&searchBy=${searchBy}`);
  }, [router, search, searchBy]);

  const handleSortMoviesClickByRating = useCallback(() => {
    router.push(`/search/?sortBy=vote_average&search=${search}&searchBy=${searchBy}`);
  }, [router, search, searchBy]);

  return (
    <SortResultContainer>
      {isSearchPage ? (
        moviesCount !== 0 && (
          <>
            <span>
              <strong>{`${moviesCount} movies found`}</strong>
            </span>
            <div>
              <span>
                <strong>Sort by: </strong>
              </span>
              <Button
                variant={VariantBtn.SortBy}
                active={sortBy === 'release_date'}
                onClick={handleSortMoviesClickByReleaseDate}
              >
                release date
              </Button>
              <Button
                variant={VariantBtn.SortBy}
                active={sortBy === 'vote_average'}
                onClick={handleSortMoviesClickByRating}
              >
                rating
              </Button>
            </div>
          </>
        )
      ) : (
        <div>
          <span>
            <strong>Films by: </strong>
          </span>
          {genres?.[0] ? <span>{genres[0]} genre</span> : <span>Unnone genre</span>}
        </div>
      )}
    </SortResultContainer>
  );
};

export default memo(SortResultsSection);
