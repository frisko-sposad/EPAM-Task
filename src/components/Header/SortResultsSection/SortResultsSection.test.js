import React from 'react';
import { shallow } from 'enzyme';
import SortResultsSection from './SortResultsSection';

describe('Films by genre.', () => {
  it('State filmsBy', () => {
    const component = shallow(<SortResultsSection filmsBy="Films by " genre="Drama genre" />);
    expect(component).toMatchSnapshot();
  });
});

describe('Films by sort', () => {
  it('State moviesFound', () => {
    const component = shallow(
      <SortResultsSection moviesFound="7 movies found" releaseDate="release date" sortBy="Sort by" rating="rating" />,
    );
    expect(component).toMatchSnapshot();
  });
});
