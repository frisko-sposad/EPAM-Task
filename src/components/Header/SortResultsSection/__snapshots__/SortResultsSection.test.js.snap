// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films by genre. State filmsBy 1`] = `
<section
  className="result-sort__container"
>
  <div>
    <span>
      <strong>
        Films by 
      </strong>
    </span>
    <span>
      Drama genre
    </span>
  </div>
</section>
`;

exports[`Films by sort State moviesFound 1`] = `
<section
  className="result-sort__container"
>
  <div>
    <span>
      <strong>
        7 movies found
      </strong>
    </span>
  </div>
  <div>
    <span>
      <strong>
        Sort by
      </strong>
    </span>
    <button
      className="btn_sort"
      type="button"
    >
      release date
    </button>
    <button
      className="btn_sort"
      type="button"
    >
      rating
    </button>
  </div>
</section>
`;
